src/Initializer/time_stepping/LtsWeights/LtsWeights.cpp

```diff
@@ -373,8 +410,246 @@ int LtsWeights::enforceMaximumDifferenceLocal(int maxDifference) {
+
+std::vector<int>& LtsWeights::getVertexWeights() { return m_vertexWeights; }
+
+std::vector<double>& LtsWeights::getImbalances() { return m_imbalances; }
+
+const std::vector<int>& LtsWeights::getCellCosts() const { return m_cellCosts; }
+
+int LtsWeights::getVertexWeightElement() const { return m_vertexWeightElement; }
+
+int LtsWeights::getVertexWeightDynamicRupture() const { return m_vertexWeightDynamicRupture; }
+
+int LtsWeights::getVertexWeightFreeSurfaceWithGravity() const {
+  return m_vertexWeightFreeSurfaceWithGravity;
+}
+
+int LtsWeights::getNcon() const { return m_ncon; }
+
+const PUML::TETPUML* LtsWeights::getMesh() const { return m_mesh; }
+
+const std::vector<int>& LtsWeights::getClusterIds() const { return m_clusterIds; }
+
+LtsWeights::LtsWeights(const LtsWeightsConfig& config) noexcept
+    : m_velocityModel(config.velocityModel), m_rate(config.rate),
+      m_vertexWeightElement(config.vertexWeightElement),
+      m_vertexWeightDynamicRupture(config.vertexWeightDynamicRupture),
+      m_vertexWeightFreeSurfaceWithGravity(config.vertexWeightFreeSurfaceWithGravity),
+      m_nodeWeightModel(nullptr), m_edgeWeightModel(nullptr) {}
+
+void LtsWeights::setVertexWeights() { m_nodeWeightModel->setVertexWeights(); }
+
+void LtsWeights::setAllowedImbalances() { m_nodeWeightModel->setAllowedImbalances(); }
+
+int LtsWeights::evaluateNumberOfConstraints() const {
+  return m_nodeWeightModel->evaluateNumberOfConstraints();
+}
+
+void LtsWeights::setEdgeWeights(std::tuple<const std::vector<idx_t>&, const std::vector<idx_t>&,
+                                           const std::vector<idx_t>&>& graph) {
+  m_edgeWeightModel->setEdgeWeights(graph);
+}
+
+LtsWeights::~LtsWeights() noexcept {
+  delete m_nodeWeightModel;
+  delete m_edgeWeightModel;
+}
+
+void LtsWeights::addWeightModels(NodeWeightModel* nwm, EdgeWeightModel* ewm) {
+  m_nodeWeightModel = nwm;
+  m_edgeWeightModel = ewm;
+}
+
+std::vector<int>& LtsWeights::getEdgeWeights() { return m_edgeWeights; }
+
+int LtsWeights::find_rank(const std::vector<idx_t>& vrtxdist, idx_t elemId){
+  assert(!vrtxdist.empty());
+  assert(elemId < vrtxdist.back());
+  assert(elemId >= 0);
+  assert(vrtxdist[0] == 0);
+
+  for (size_t i = 0; i < vrtxdist.size()-1; i++){
+    if (elemId >= vrtxdist[i] && elemId < vrtxdist[i+1])
+    {
+      return i;
+    }
+  }
+
+  throw std::runtime_error("element id is not in valid range!");
+}
+
+
+std::vector<std::unordered_map<idx_t, int>>
+LtsWeights::exchangeGhostLayer(std::tuple<const std::vector<idx_t>&, const std::vector<idx_t>&,
+                                          const std::vector<idx_t>&>& graph) {
+  // fails with intel compiler, mpi standard says use mpi size of
+  // static_assert(sizeof(idx_t) <= sizeof(MPI_LONG_LONG_INT));
+
+  const std::vector<idx_t>& vrtxdist = std::get<0>(graph);
+  const std::vector<idx_t>& xadj = std::get<1>(graph);
+  const std::vector<idx_t>& adjncy = std::get<2>(graph);
```

Comment:
I will need to think of names for xadj and adjncy, I have taken the names as it is from the METIS library. I haven´t seen nice names for these elements in the metis documentation. But if any user wants to change these vector or work with them they will need to have some grasp of the metis library and its data format and therefore keeping the names as they are can be a considered (I had thought)? 
If you think otherwise I will try to find some more descriptive names

Reply:
ok, then probably that's fine as it is.